src/drt/src/pa/FlexPA_acc_point.cpp

```diff
@@ -1166,27 +1158,58 @@ bool FlexPA::EnoughAccessPoints(
 
   if (is_std_cell_pin
       && n_sparse_access_points >= router_cfg_->MINNUMACCESSPOINT_STDCELLPIN) {
-    enough_sparse_acc_points = true;
+    return true;
   }
   if (is_macro_cell_pin
       && n_sparse_access_points
              >= router_cfg_->MINNUMACCESSPOINT_MACROCELLPIN) {
-    enough_sparse_acc_points = true;
+    return true;
   }
+  return false;
+}
 
+bool FlexPA::EnoughPointsFarFromEdge(
+    std::vector<std::unique_ptr<frAccessPoint>>& aps,
+    frInstTerm* inst_term)
+{
+  const int far_from_edge_requirement = 1;
   Rect cell_box = inst_term->getInst()->getBBox();
+  int total_far_from_edge = 0;
   for (auto& ap : aps) {
     const int colision_dist
         = design_->getTech()->getLayer(ap->getLayerNum())->getWidth() * 2;
```

Comment:
@bnmfw Could you clarify the rationale behind multiplying by 2 here? The current collision_dist calculation seems somewhat conservative, which might result in unnecessarily restrictive pin access